tile/platform/local_machine/scheduler.cc

```diff
@@ -382,6 +382,14 @@ void ValidateSchedule(const tile::proto::Program& program, const lang::KernelLis
   }
 }
 
+std::size_t TotalAllocSize(const schedule::Schedule& schedule) {
+  std::size_t total_size = 0;
+  for (const auto& alloc : schedule.allocs) {
+    total_size += alloc.byte_size;
```

Comment:
You need to take into account the device's underlying allocation alignment in this computation.  `AlignUp` from `fifo_scheduler.cc` might be useful.